src/main/java/org/prebid/server/analytics/reporter/greenbids/GreenbidsAnalyticsReporter.java

```diff
@@ -0,0 +1,323 @@
+import org.prebid.server.analytics.reporter.greenbids.model.GreenbidsConfig;
+import org.prebid.server.analytics.reporter.greenbids.model.HttpUtil;
+import org.prebid.server.analytics.reporter.greenbids.model.JsonUtil;
+import org.prebid.server.analytics.reporter.greenbids.model.MediaTypes;
+import org.prebid.server.auction.model.AuctionContext;
+import org.prebid.server.auction.model.BidRejectionReason;
+import org.prebid.server.auction.model.BidRejectionTracker;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.proto.openrtb.ext.request.ExtImpPrebid;
+import org.prebid.server.proto.openrtb.ext.request.ExtRequest;
+import org.prebid.server.proto.openrtb.ext.request.ExtRequestPrebid;
+import org.prebid.server.proto.openrtb.ext.request.ExtStoredRequest;
+import org.prebid.server.proto.openrtb.ext.response.seatnonbid.NonBid;
+import org.prebid.server.proto.openrtb.ext.response.seatnonbid.SeatNonBid;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+public class GreenbidsAnalyticsReporter implements AnalyticsReporter {
+
+    private static final String ANALYTICS_SERVER = "https://a.greenbids.ai/";
+    public String pbuid;
+    public Double greenbidsSampling;
+    public Double exploratorySamplingSplit;
+    private final GreenbidsConfig greenbidsConfig;
+    private final JacksonMapper jacksonMapper;
+    public Boolean isSampled;
+    public String greenbidsId;
+    public String billingId;
+
+    public GreenbidsAnalyticsReporter(
+            GreenbidsAnalyticsProperties greenbidsAnalyticsProperties,
+            JacksonMapper jacksonMapper
+    ) {
+        this.pbuid = Objects.requireNonNull(greenbidsAnalyticsProperties.getPbuid());
+        this.greenbidsSampling = greenbidsAnalyticsProperties.getGreenbidsSampling();
+        this.exploratorySamplingSplit = 0.9;
+        this.greenbidsConfig = GreenbidsConfig.of(
+                greenbidsAnalyticsProperties.getPbuid(),
+                greenbidsAnalyticsProperties.getGreenbidsSampling()
+        );
+        this.jacksonMapper = Objects.requireNonNull(jacksonMapper);
+        this.isSampled = null;
+        this.greenbidsId = null;
+        this.billingId = null;
+    }
+
+    public CommonMessage createCommonMessage(AuctionContext auctionContext, Long auctionElapsed) {
+        return new CommonMessage(
+                auctionContext,
+                greenbidsConfig,
+                auctionElapsed,
+                greenbidsId,
+                billingId
+        );
+    }
+
+    private String getGpid(ObjectNode impExt) {
+        return Optional.ofNullable(impExt)
+                .map(ext -> ext.get("prebid"))
+                .map(JsonNode::asText)
+                .orElse(null);
```

Comment:
Here it was actually a typo: to extract the `gpid` in ObjectNode `ext` instead of searching for the key `prebid` we need to search for the key `gpid`. I've fixed it https://github.com/prebid/prebid-server-java/pull/3096#discussion_r1601215748